sources/Inputs/src/NetworkManager.cpp

```diff
@@ -127,6 +112,16 @@ void NetworkManager::buildTree() {
         nodes_[bus->getID()]->busBarSections.emplace_back(busBarSection);
       }
     }
+  }
+  // And then we update others elements
+  for (const auto &networkVL : voltageLevels) {
+    const auto &shunts = networkVL->getShuntCompensators();
+    std::unordered_map<Node::NodeId, std::vector<Shunt>> shuntsMap;
+    for (const auto &shunt : shunts) {
+      updateConditioningStatus(shunt);
+      // We take into account even disconnected shunts as dynamic models may aim to connect them
+      (shuntsMap[shunt->getBusInterface()->getID()]).push_back(std::move(Shunt(shunt->getID())));
+    }
 
```

Comment:
Not sure I get why you've to re-iterate over `voltageLevels`?

Reply:
The first loop initializes the nodes, the second one uses them to initialize components. It was not required before, but it is now to retrieve the regulated nodes in case of distant regulation.